python_modules/dagster-graphql/dagster_graphql/client/client.py

```diff
@@ -396,3 +400,189 @@ def terminate_run(self, run_id: str):
+                        {
+                            "assetKey": metric.asset_key,
+                            "assetGroup": metric.asset_group,
+                            "metricValues": [
+                                {
+                                    "metricValue": metric_def.metric_value,
+                                    "metricName": metric_def.metric_name,
+                                }
+                                for metric_def in metric.metrics
+                            ],
+                        }
+                    )
+            elif isinstance(metric, JobMetric):
+                if key not in metric_graphql_inputs.keys():
+                    metric_graphql_inputs[key] = {
+                        "runId": metric.run_id,
+                        "stepKey": metric.step_key,
+                        "codeLocationName": metric.code_location_name,
+                        "repositoryName": metric.repository_name,
+                        "assetMetricDefinitions": [],
+                        "jobMetricDefinitions": [
+                            {
+                                "metricValues": [
+                                    {
+                                        "metricValue": metric_def.metric_value,
+                                        "metricName": metric_def.metric_name,
+                                    }
+                                    for metric_def in metric.metrics
+                                ],
+                            }
+                        ],
+                    }
+                else:
+                    metric_graphql_inputs[key]["jobMetricDefinitions"].append(
+                        {
+                            "metricValues": [
+                                {
+                                    "metricValue": metric_def.metric_value,
+                                    "metricName": metric_def.metric_name,
+                                }
+                                for metric_def in metric.metrics
+                            ],
+                        }
+                    )
+            else:
+                raise DagsterGraphQLClientError(
+                    "InvalidMetricTypeError",
+                    f"Metric {metric} is not of type AssetMetric or JobMetric",
+                )
+
+        res_data: Dict[str, Dict[str, Any]] = self._execute(
+            PUT_CLOUD_METRICS_MUTATION,
+            {"metrics": metric_graphql_inputs.values()},
+        )
+
+        store_metric_result: Dict[str, Any] = res_data["createExternalMetrics"]
+        store_metric_result_type: str = store_metric_result["__typename"]
+        if store_metric_result_type == "CreateExternalMetricsSuccess":
+            return
+
+        raise DagsterGraphQLClientError(store_metric_result_type, store_metric_result["message"])
+
+    @experimental
+    @public
+    def put_context_metrics(
```

Comment:
nit - slight preference for something like `put_metrics_from_context` or `_with_context` since I feel like that more clearly conveys the role of the context here